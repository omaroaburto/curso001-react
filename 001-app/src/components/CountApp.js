import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CountApp = ({value}) => {
    //handAdd
    const [count, setCount] = useState(value);
    const handleAdd = () =>{ 
        //forma 1
        setCount(count+1);
        //forma 2
        //setCount((c)=> c+1);
    } 
    const handleSubtract = () =>{ 
        //forma 1
        if(count!==0)
            setCount(count-1); 
    } 
    const handleReset = () =>{ 
        //forma 1
        setCount(value); 
    } 
    return (
        <>
            <h1>CounterApp</h1>
            <span>{count}</span> 
            <br/>
            <button onClick={handleAdd}>+1</button> 
            <button onClick={handleSubtract}>-1</button> 
            <button onClick={handleReset}>Reset</button> 
        </>
    );
}

CountApp.propTypes = {
    value: PropTypes.number
}
CountApp.defaultProps = {
    value: 10
}
export default CountApp
