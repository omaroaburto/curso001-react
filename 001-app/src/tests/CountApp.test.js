import React from 'react';  
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import CountApp from '../components/CountApp';

describe('Son las pruebas del archivo CounterApp.test.js',()=>{
    let wrapper
    //se ejecta antes de cada test
    beforeEach(()=>{
        wrapper =  shallow(<CountApp/>); 
    });

    //test 1
    test('Prueba de componente <CountApp />',()=>{
        expect(wrapper).toMatchSnapshot();
    });

    //test 2
    test('Prueba de <CountApp/> con props', ()=>{
        const value = 100;
        const wrapper =  shallow(<CountApp value = {value}/>); 
        expect(wrapper).toMatchSnapshot();
    });

    //test 3
    test('Prueba <CountApp/> el botón +1 debe aumentar el valor en 1', ()=>{
        wrapper.find('button').at(0).simulate('click');
        const countText =  wrapper.find('span').text().trim();
        expect(countText).toBe('11');
    });

    //test 4
    test('Prueba <CountApp/> el botón -1 debe disminuir el valor en 1', ()=>{
        wrapper.find('button').at(1).simulate('click');
        const countText =  wrapper.find('span').text().trim(); 
        expect(countText).toBe('9');
    });
});