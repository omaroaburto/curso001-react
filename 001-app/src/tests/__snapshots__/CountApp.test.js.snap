// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Son las pruebas del archivo CounterApp.test.js Prueba de <CountApp/> con props 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <span>
    100
  </span>
  <br />
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;

exports[`Son las pruebas del archivo CounterApp.test.js Prueba de componente <CountApp /> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <span>
    10
  </span>
  <br />
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
